### ISSUE TITLE
No http headers in environment with preheated kernels

### ISSUE THREAD

**Issue by sradomski** [voila-dashboards/voila#1113]
### Problem

When using the preheated kernel mode, HTTP headers from http_header_envs are no longer forwarded into the environment.

We do set some HTTP headers via a reverse proxy e.g. to make voila aware of the current SSO user and have voila forward them as environmental variables. E.g. passing the command-line option `--VoilaConfiguration.http_header_envs='REQ_HEADERS_Email'` makes its value available as `os.environ['HTTP_REQ_HEADERS_EMAIL']`.

However, when serving the notebooks in preheated mode, those environmental variables are no longer available.

### Proposed Solution

Set the environmental variable in the kernels environment, maybe via an approach similar to `get_query_string()`?

**Comment by maartenbreddels** [comment]
See https://github.com/voila-dashboards/voila/pull/1109 :)

**Comment by sradomski** [comment]
😀 thank you!

I suppose these changes will be part of the 0.3.4 release? Is there any indication as to when we may find a new package in pip / conda? Trying to decide on whether to use voila from git or to wait for a new release.

**Comment by trungleduc** [comment]
@sradomski yes, this PR will be part of 0.3.4. While waiting for the next release, you can take a look at the setting of [preheated kernel](https://voila.readthedocs.io/en/stable/customize.html#preheated-kernels)
```python
# voila.json
      
      ...
      "kernel_pools_config": {
         # Setting for `voila.ipynb` notebook
         "voila.ipynb": {
            "pool_size": 3, # Size of pool
            "kernel_env_variables": { # The environment variables used to start kernel for `voila.ipynb`
               "foo2": "bar2"
            }
         }
      ...
```
You can set environmental variables of preheated kernel with `kernel_env_variables` key
